### ISSUE TITLE
Error when clicking on repo name

### ISSUE THREAD

**Issue by stanspotts** [Magenic/AzureDevOpsGitOrganization#7]
I installed the extension, and added a root folder and a subfolder, and added one repo under the subfolder. When I clicked on the repo, I go this:

```
{
"$id": "1",
"innerException": null,
"message": "The extension 'Magenic.ado-source-cat' doesn't contain the requested asset type 'undefined' in version '0.1.56'.",
"typeName": "Microsoft.VisualStudio.Services.Gallery.WebApi.ExtensionAssetNotFoundException, Microsoft.VisualStudio.Services.Gallery.WebApi",
"typeKey": "ExtensionAssetNotFoundException",
"errorCode": 0,
"eventId": 3000
}
```

I imagine that wasn't the expected result :).

**Comment by stanspotts** [comment]
Using Azure DevOps Server Version 17.143.28621.4, BTW.

**Comment by Bowman74** [comment]
No, probably not the expected result, perhaps some difference with the on-prem version. In the new version there is a way to examine the json of the treeview. Can you get that and paste it here?

It should navigate you to the repo when you click on the repo leaf but I'm thinking the URL in there is ... not what we expect.

**Comment by stanspotts** [comment]
No URL's in the json, and I just removed the old version and installed today from the Marketplace:
`[{"name":"RtTop","nodeType":"folder","children":[{"name":"Life Sciences","nodeType":"folder","children":[{"name":"LS_LSPAdmin","nodeType":"repository"},{"name":"LS_LDR","nodeType":"repository"},{"name":"LS_PDR","nodeType":"repository"},{"name":"LS_LDP","nodeType":"repository"},{"name":"LS_UDI","nodeType":"repository"},{"name":"LS_GRDM","nodeType":"repository"},{"name":"LS_USL","nodeType":"repository"},{"name":"LS_LDRRepo","nodeType":"repository"}]},{"name":"Patents","nodeType":"folder","children":[{"name":"PA_AutomationTools","nodeType":"repository"},{"name":"PA_Classification","nodeType":"repository"},{"name":"PA_ePM_eQC","nodeType":"repository"},{"name":"PA_DBProjects","nodeType":"repository"},{"name":"PA_CofC","nodeType":"repository"}]},{"name":"Infrastructure","nodeType":"folder"},{"name":"Tools","nodeType":"folder"}]}]`

**Comment by Bowman74** [comment]
@stanspotts That json is definitely invalid. If you manually add the urls does it work? 

The question is how did this json get created in the first place?

**Comment by Bowman74** [comment]
Also are there any special characters in your repo names?

By repo names I really mean the URL to the repo, it doesn't look like there is anything in the repo name itself other than a _.

**Comment by Bowman74** [comment]
It could be that the API of the on-premise version is different from Azure DevOps and it can't find the URL. In such a case manually editing the Json should do the job.

**Comment by stanspotts** [comment]
I simply added them via the UI.


—Stan

On Sep 20, 2019, at 11:31 AM, Kevin E. Ford <notifications@github.com> wrote:

﻿

@stanspotts<https://github.com/stanspotts> That json is definitely invalid. If you manually add the urls does it work?

The question is how did this json get created in the first place?

**Comment by stanspotts** [comment]
What should it look like?

—Stan

On Sep 22, 2019, at 3:07 PM, Kevin E. Ford <notifications@github.com> wrote:

﻿

It could be that the API of the on-premise version is different from Azure DevOps and it can't find the URL. In such a case manually editing the Json should do the job.
